src/generators/SpirvShaderGenerator.cpp

```diff
@@ -530,6 +612,44 @@ void CSpirvShaderGenerator::Generate()
 					StoreToSymbol(dstRef, resultId);
 				}
 				break;
+			case CShaderBuilder::STATEMENT_OP_TOUCHAR:
+				{
+					auto src1Id = LoadFromSymbol(src1Ref);
+					auto resultId = AllocateId();
+					switch(src1Ref.symbol.type)
+					{
+					case CShaderBuilder::SYMBOL_TYPE_FLOAT4:
+						WriteOp(spv::OpConvertFToU, m_uchar4TypeId, resultId, src1Id);
+						break;
+					case CShaderBuilder::SYMBOL_TYPE_UINT4:
+						WriteOp(spv::OpUConvert, m_uchar4TypeId, resultId, src1Id);
```

Comment:
if we use `OpBitcast` here, the resulting value will be treated as 32bit value, even if RenderDoc GLSL dism view would show the data type as uchar/ubyte